Doc/Manual/Doxygen.html

```diff
@@ -0,0 +1,1735 @@
+"http://www.stack.nl/~dimitri/doxygen/">Doxygen.</A> They can be
+present in your main interface file or any header file that it
+imports.  It is advised that you are certain your comments compile
+properly with Doxygen before you try to translate them.  Doxygen
+itself is a deeper tool and can provide you better feedback for
+correcting any syntax errors that may be present.  Please look at
+Doxygen's <A HREF =
+"http://www.stack.nl/~dimitri/doxygen/docblocks.html"> Documenting the
+code</A> for proper specifications for comment format.  However, SWIG's
+Doxygen parser will still point you most of errors and warnings found
+in comments (like unterminated strings or missing ending tags).
+</p>
+
+<div class="code"><pre>
+/*! This is describing class Shape
+ \author Bob
+ */
+
+class Shape {
+</pre></div>
+
+<p>
+Currently, the whole subset of Doxygen comment styles is supported
+(See <A HREF = "http://www.stack.nl/~dimitri/doxygen/docblocks.html">
+Documenting the code</A>). Here they are:
+
+<div class="code"><pre>
+/**
+ * Javadoc style comment, multiline
+ */
+/*!
+ * QT-style comment, multiline
+ */
+/**
+ Any of the above, but without intermediate *'s
+ */
+/// Single-line comment
+//! Another single-line comment
+</pre></div>
+</p>
+
+<p>
+Also any of the above with '&lt;' added after comment-starting symbol,
+like <i>/**&lt;, /*!&lt;, ///&lt;, </i> or <i> //!&lt;</i> will be
+treated as post-comment and will be assigned to the node before the
+comment.
+<br>
+
+Any number of '*' or '/' in Doxygen comment is considered to be a
+separator and is not included in final comment, so you may safely use
+comments like <i>/*********/</i> or <i>//////////</i>.
+</p>
+
+<p>
+Please note, as SWIG parses input file by itself with strict grammar,
+there is only a limited support for various cases of comment placement
+in the file.
+<br>
+Comments can be placed before C/C++ expressions on separate lines:
+</p>
+
+<div class="code"><pre>
+/**
+ * Some comment
+ */
+void someOtherFunction();
+/**
+ * Some comment
+ */
+void someFunction();
+
+class Shape {
+  /*
+   * Calculate the area in cm^2
+   */
+  int getArea();
+}
+</pre></div>
+
+<p>
+After C/C++ expressions at the end of the line:
+</p>
+
+<div class="code"><pre>
+int someVariable = 9; ///&lt; This is a var holding magic number 9
+void doNothing(); ///&lt; This does nothing, nop
+</pre></div>
+
+<p>
+and in some special cases, like function parameter comments:
+</p>
+
+<div class="code"><pre>
+void someFunction(
+         int a ///&lt; Some parameter 
+     );
+</pre></div>
+
+<p>
+or enum element comments:
+</p>
+
+<div class="code"><pre>
+enum E_NUMBERS
+{
+    EN_ZERO, ///&lt; The first enum item, gets zero as it's value
+    EN_ONE, ///&lt; The second, EN_ONE=1
+    EN_THREE
+};
+</pre></div>
+
+<p>
+Just remember, if SWIG shows syntax error parsing the file because of
+your comment, try to move it in some other, 'safer' place as described
+above.
+<br>
```

Comment:
@vadz, isn't this comment out of date now since you impoved the implementation? Would a doxygen comment ever introduced a syntax error?

Reply:
I _hope_ this shouldn't happen any more, but, to be honest, I'm not 100% sure about this. But I think that if it does still happen, it should be considered a bug, and so this part of the documentation should still be removed.